src/realm/object-store/c_api/query.cpp

```diff
@@ -177,15 +179,14 @@ struct QueryArgumentsAdapter : query_parser::Arguments {
 } // namespace
 
 static Query parse_and_apply_query(const std::shared_ptr<Realm>& realm, ConstTableRef table,
-                                   DescriptorOrdering& ordering, const char* query_string, size_t num_args,
-                                   const realm_value_t* args)
+                                   util::bind_ptr<DescriptorOrdering>& ordering, const char* query_string,
```

Comment:
I don't think we need this parameter anymore. We can just return the Query and the caller can call `query.get_ordering()` on it as needed. Or better yet, we could move this into the constructor of realm_query_t since it looks like most of the time we extract this bind_ptr only to immediately move it into the constructor of realm_query_t without any further actions on it.

Reply:
:100: